databricks_asset_bundles/00.create_source_data.py

```diff
@@ -0,0 +1,100 @@
+from pyspark.sql import SparkSession
+from pyspark.sql.types import *
+
+# Create SparkSession
+spark = SparkSession.builder.getOrCreate()
+
+import pandas as pd
+import numpy as np
+from pyspark.sql.functions import current_timestamp, to_utc_timestamp
+from nyctaxi.config import ProjectConfig
+
+
+# Load configuration
+config = ProjectConfig.from_yaml(config_path="project_config.yml")
+
+catalog_name = config.catalog_name
+schema_name = config.schema_name
+
+# Load train and test sets
+train_set = spark.table(f"{catalog_name}.{schema_name}.train_set_an").toPandas()
+test_set = spark.table(f"{catalog_name}.{schema_name}.test_set_an").toPandas()
+combined_set = pd.concat([train_set, test_set], ignore_index=True)
+num_rows = len(combined_set)
+
+print(combined_set)
+#existing_ids = set(int(id) for id in combined_set['Id'])
+
+# Define function to create synthetic data without random state
+def create_synthetic_data(df, num_rows=100):
+    synthetic_data = pd.DataFrame()
+    
+    for column in df.columns:
+        if pd.api.types.is_numeric_dtype(df[column]):
+            mean, std = df[column].mean(), df[column].std()
+            synthetic_data[column] = np.random.normal(mean, std, num_rows)
+        
+        elif pd.api.types.is_categorical_dtype(df[column]) or pd.api.types.is_object_dtype(df[column]):
+            synthetic_data[column] = np.random.choice(df[column].unique(), num_rows, 
+                                                      p=df[column].value_counts(normalize=True))
+        
+        #elif pd.api.types.is_datetime64_any_dtype(df[column]):
+            #min_date, max_date = df[column].min(), df[column].max()
+            #if min_date < max_date:
+                #synthetic_data[column] = pd.to_datetime(
+                    #np.random.randint(min_date.value, max_date.value, num_rows)
+                #)
+            #else:
+                #synthetic_data[column] = [min_date] * num_rows
+        
+        else:
+            synthetic_data[column] = np.random.choice(df[column], num_rows)
+    
+    #new_ids = []
+    #i = max(existing_ids) + 1 if existing_ids else 1
```

Comment:
how do you ensure that new dataset will have unique ids?

Reply:
To clarify: I use the New York City Taxi Set that does not have IDs. I instead use postal_code as a key column, which is not unique. I know it is not perfect, I just use it for representative matters